### DISCUSSION TITLE
Search bar component

### DISCUSSION THREAD

**Discussion by ivandimitrov8080** [saadeghi/daisyui#2270] [Ideas / feature requests]
I'd like to be able to easily add a search bar to a website similar to what you have on daisyui.com:
![image](https://github.com/saadeghi/daisyui/assets/80976541/17eb1a5b-557f-4b0f-8f83-4ad930a6db28)

**Comment by saadeghi** [comment]
daisyUI is just CSS. for a search input with suggestions and autocomplete, you would need JS.  
I used [`svelte-typeahead`](https://github.com/metonym/svelte-typeahead) for the site.

**Reply by ivandimitrov8080** [reply]
My reply is below, but I wanted to ask something else.

How come daisyui doesn't work that well when you disable js in the browser? Does it have absolutely 0 JS or just a little bit for styling?

**Reply by saadeghi** [reply]
There's no JS in daisyUI. It just adds class names to the Tailwind CSS.  
The only thing that needs JS is the `<dialog>` element which needs `onclick="my_modal.showModal()"` because that's how this HTML element works and that 1 line is in the examples here:  
https://daisyui.com/components/modal/#method-1-using-dialog-element-recommended

daisyUI itself has no JS and doesn't require JS.

**Comment by ivandimitrov8080** [comment]
Would be nice to have the CSS and HTML written and leave the user to implement the actual logic by adding an array of elements to where the results are supposed to go.

**Reply by saadeghi** [reply]
There are many packages on NPM and devs can use them. Usually there's no need to implement the logic by yourself. You just need to add daisyUI class names (or any custom style to their HTML) however these packages have their own HTML and there's no point of I provide some HTML for that in daisyUI.
